src/nifreeze/utils/iterators.py

```diff
@@ -53,28 +56,36 @@ def linear_iterator(size: int | None = None, **kwargs) -> Iterator[int]:
     if size is None:
         raise TypeError("Cannot build iterator without size")
 
-    return iter(range(size))
+    return (s for s in range(size))
 
 
 def random_iterator(size: int | None = None, **kwargs) -> Iterator[int]:
     """
     Traverse the dataset volumes randomly.
 
+    If the `seed` key is present in the keyword arguments, initializes the seed
+    of Python's `random` pseudo-random number generator library with the given
+    value. Specifically, if `False`, `None` is used as the seed; it `True`, a
+    default seed value is used.
+
     Parameters
     ----------
-    size : :obj:`int`
-        Number of volumes in the dataset
-        (for instance, the number of orientations in a DWI)
-    seed : :obj:`int` or :obj:`bool` or :obj:`bool` or ``None``
+    size : :obj:`int` or ``None`` (optional)
+        Number of volumes in the dataset.
+    bvalues : :obj:`list` (optional, keyword argument)
```

Comment:
Same comment as above, not sure what Python's standard way to document these is.